plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/listener/SpringEvent.java

```diff
@@ -0,0 +1,36 @@
+package org.hotswap.agent.plugin.spring.listener;
+
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+
+import java.util.EventObject;
+
+/**
+ * Spring event.
+ * It is used to notify the plugin that a bean definition has changed.
+ * Spring boot and Spring should send some events each other, the event and listener are used to satisfy this requirement.
+ *
+ * @param <T> the event type
+ */
+public abstract class SpringEvent<T> extends EventObject {
```

Comment:
One thought, should we really need to introduce the concept of SpringEvent for now? I think we can consider to introduce this concept when it is really necessary in the future.